File: spambot/SPAMFramework/Packet/HandOverPacket.cpp
#include "HandOverPacket.h"

HandOverPacket::HandOverPacket(int my_stack_size, int villain_stack_size,
			int num_board_cards, std::vector<Card*> board_cards, int num_last_actions,
			std::vector<std::string> last_actions, double time_bank) {

	m_stack_size = my_stack_size;
	m_villain_stack_size = villain_stack_size;
	m_num_board_cards = num_board_cards;
	m_board_cards = board_cards;
	m_num_last_actions = num_last_actions;
	m_last_actions = last_actions;
	m_time_bank = time_bank;
}

void HandOverPacket::UpdateGameState(GameState *gs) {
	gs->SetMyStackSize(m_stack_size);
	gs->SetVillainStackSize(m_villain_stack_size);
	gs->SetCurrentStreet(GameState::HANDOVER);
	gs->SetBoardCards(m_board_cards);
	
	// TODO: Need to separate the two action histories
	gs->GetSelfHistory()->SetLastActions(m_last_actions);
	gs->GetVillainHistory()->SetLastActions(m_last_actions);
	
	gs->SetTimeLeft(m_time_bank);
}

File: java_bot/src/pokerbots/spam_framework/packet/HandOverPacket.java
package pokerbots.spam_framework.packet;

import java.util.ArrayList;

import pokerbots.spam_framework.Card;
import pokerbots.spam_framework.GameState;

public class HandOverPacket extends Packet {

	private int myStackSize;
	private int villainStackSize;
	private int numBoardCards;
	private ArrayList<Card> boardCards;
	private int numLastActions;
	private ArrayList<String> lastActions;
	private double timeBank;
	
	public HandOverPacket(int myStackSize, int villainStackSize,
			int numBoardCards, ArrayList<Card> boardCards, int numLastActions,
			ArrayList<String> lastActions, double timeBank) {
		
		this.myStackSize = myStackSize;
		this.villainStackSize = villainStackSize;
		this.numBoardCards = numBoardCards;
		this.boardCards = boardCards;
		this.numLastActions = numLastActions;
		this.lastActions = lastActions;
		this.timeBank = timeBank;
	}
	
	@Override
	public void updateGameState(GameState gameState) {
		gameState.setMyStackSize(myStackSize);
		gameState.setVillainStackSize(villainStackSize);
		gameState.setCurrentStreet(GameState.HANDOVER);
		gameState.setBoardCards(boardCards);
		
		// TODO: Need to separate the two action histories
		gameState.getSelfHistory().setLastActions(lastActions);
		gameState.getVillainHistory().setLastActions(lastActions);
		
		gameState.setTimeLeft(timeBank);
	}

}
